### ISSUE TITLE
Crash on collapse all

### ISSUE THREAD

**Issue by alexarice** [sd-visualiser/sd-visualiser#72]
Crashed when running collapse all on ad.chil

```
thread 'shape' panicked at 'called `Option::unwrap()` on a `None` value', /home/alex/sd-visualiser/sd-core/src/monoidal/wired_graph.rs:270:79
```

**Comment by alexarice** [comment]
1c003e63026b9ddabc4844623b3add080112be76 is the first bad commit
commit 1c003e63026b9ddabc4844623b3add080112be76
Author: Calin Tataru <calintat@gmail.com>
Date:   Mon Sep 18 14:09:18 2023 +0100

    Integrate collapse graphs in rendering pipeline

 sd-core/src/hypergraph/adapter/collapse.rs | 129 ++++++++++++++++++++++++++++-
 sd-core/src/hypergraph/adapter/cut.rs      |  13 ++-
 sd-core/src/interactive.rs                 | 108 +++++++++++++++++++++---
 sd-graphics/src/common.rs                  |   3 +-
 sd-graphics/src/layout.rs                  |  26 ++----
 sd-graphics/src/render.rs                  |  37 ++-------
 sd-graphics/src/renderable.rs              |  58 +++++++++----
 sd-graphics/src/shape.rs                   |  39 +++------
 sd-gui/src/graph_ui.rs                     |  35 +++-----
 sd-gui/src/selection.rs                    |  39 +++------
 sd-gui/src/shape_generator.rs              |  18 ++--
 11 files changed, 325 insertions(+), 180 deletions(-)

**Comment by calintat** [comment]
This seems to be fixed by 40b6ffbe21cdf20739e38e70a8351cd8c8b96bda
